File: 10-233B-Non-square Equation/03-B. Non-square Equation.py
import  math
def S(n):
    cpt = 0     
    while n  != 0:     
        cpt +=  n  %  10     
        n = n // 10   
    return cpt
S(110) 
cpt2 = 0    
n =  int(input())          
sqrt_  = math.sqrt (n)      
start = max (1 , int(sqrt_ - 162))   
end = int(sqrt_)     
for x in range (start , end+1) :        
    target1 =  S(x)  
    result  = (x * x) + target1 * x    
    if result  ==  n :     
        print (x)      
        cpt2 += 1   
        break      
if cpt2  == 0  :
    print(-1) 

File: 10-233B-Non-square Equation/03-B-Non-square Equation.java
import java.util.Scanner;

public class Main {
    public static long S(long n) {
        long cpt = 0;
        while (n != 0) {
            cpt += n % 10;
            n /= 10;
        }
        return cpt;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long n = sc.nextLong();

        long sqrt_ = (long)Math.sqrt(n);
        long start = Math.max(1, sqrt_ - 162);
        long end = sqrt_;

        boolean found = false;
        for (long x = start; x <= end; x++) {
            if (x*x + S(x)*x == n) {
                System.out.println(x);
                found = true;
                break;
            }
        }

        if (!found)
            System.out.println(-1);
    }
}
